crates/sparrow-journal/src/checkpoints.rs

```diff
@@ -0,0 +1,254 @@
+use std::path::PathBuf;
+use std::str::FromStr;
+
+use arrow_array::cast::AsArray;
+use arrow_array::types::{Int64Type, TimestampNanosecondType};
+use arrow_array::RecordBatch;
+use arrow_schema::SchemaRef;
+use error_stack::{IntoReport, ResultExt};
+use futures::stream::BoxStream;
+use futures::{StreamExt, TryStreamExt};
+use sparrow_batch::{Batch, RowTime};
+
+use crate::error::Error;
+
+/// Manages the checkpointed files.
+///
+/// We could use Parquet files for check-pointing but we instead use
+/// Arrow IPC files, since we believe they should be easier to read.
```

Comment:
Can you elaborate on this for me? Why are Arrow IPC files > parquet

Reply:
Arrow IPC files write data out using the same format as Arrow, so they can largely be mapped into memory directly, while Parquet requires some things like decompression and changing formats. So, we'd likely expect:

- Arrow IPC may be faster for read/write.
- Parquet may have wider support as an archival format (eg., on object store)